File: targets/swift/Sources/App/Vuln34Cors.swift
import Vapor

func registerVuln34(_ app: Application) {
    app.get("vuln34", "data") { req -> Response in
        let resp = Response(status: .ok, body: .init(string: "{\"secret\":\"account-balance-42000\"}"))
        if let origin = req.headers.first(name: "Origin") {
            // VULN:VULN-34:CWE-942:config arbitrary Origin reflected + credentials allowed
            resp.headers.replaceOrAdd(name: "Access-Control-Allow-Origin", value: origin)
            resp.headers.replaceOrAdd(name: "Access-Control-Allow-Credentials", value: "true")
        }
        return resp
    }
}

File: targets/python-web/vulns/vuln34_cors.py
from flask import Blueprint, request, make_response
bp = Blueprint("vuln34", __name__)

@bp.get("/vuln34/data")
def data():
    resp = make_response({"secret": "account-balance-42000"})
    origin = request.headers.get("Origin")
    if origin:
        # VULN:VULN-34:CWE-942:config arbitrary Origin reflected + credentials allowed
        resp.headers["Access-Control-Allow-Origin"] = origin
        resp.headers["Access-Control-Allow-Credentials"] = "true"
    return resp
